packages/nft/src/NFTHandler.ts

```diff
@@ -121,31 +130,103 @@ export class NFTHandler {
   }
 
   /**
-   * It takes a tokenId, and returns a transaction object
-   * @param {NFT721Manager} nftManager - NFT721Manager - The NFT721Manager instance that will be used
-   * to set the NFT tokenURI.
-   * @param {StorageType} storage - StorageType - The storage to save the NFT metadata.
-   * @param {NFT_METADATA_FORMAT} nftFormat - The format of the NFT metadata.
-   * @param {NFTData} nftData - The data of the NFT you want to set.
-   * @returns The transaction object
+   * It uploads the image and/or metadata to the storage, and then calls the `setTokenURI` function on the
+   * NFT manager
+   * @param {NFTHandlerChangeParams}  - NFTHandlerChangeParams dict
+   * @return {Promise<interfaces.ExecuteResponse>} returns a promise with an ExecuteResponse.
    */
   static async change({
     nftManager,
     storage,
     nftFormat,
     tokenId,
     nftMetadata,
-    image
+    image,
+    value
   }: NFTHandlerChangeParams): Promise<interfaces.ExecuteResponse> {
     if (!nftManager.isActive) {
       throw MUST_ACTIVATE
     }
 
     if (!nftManager.implements('setTokenURI', ['uint256', 'string'])) {
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      throw SET_TOKEN_URI_NOT_IMPLEMENTED(nftManager.contractAddr!)
+      throw FUNCTION_NOT_IMPLEMENTED(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        nftManager.contractAddr!,
+        'setTokenURI(uint256,string)'
+      )
     }
 
+    const tokenUri = await NFTHandler.uploadData(
+      nftFormat,
+      storage,
+      image,
+      nftMetadata
+    )
+
+    return await nftManager.execute({
+      method: 'setTokenURI(uint256,string)',
+      value,
+      params: [tokenId, tokenUri]
+    })
+  }
+
+  /**
+   * It uploads the image and/or metadata to the storage, and then calls the `safeMint`
+   * function on the NFT manager
+   * @param {NFTHandlerCreateParams}  - NFTHandlerCreateParams dict
+   * @return {Promise<interfaces.ExecuteResponse>} returns a promise with an ExecuteResponse.
+   */
+  static async create({
+    nftManager,
+    storage,
+    nftFormat,
+    address,
+    tokenId,
+    nftMetadata,
+    image,
+    value
+  }: NFTHandlerCreateParams): Promise<interfaces.ExecuteResponse> {
+    if (!nftManager.isActive) {
+      throw MUST_ACTIVATE
+    }
+    const params = tokenId
+      ? ['address', 'uint256', 'string']
+      : ['address', 'string']
+    if (!nftManager.implements('safeMint', params)) {
+      throw FUNCTION_NOT_IMPLEMENTED(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        nftManager.contractAddr!,
+        `safeMint(${params.join(',')})`
+      )
+    }
+    const uri = await NFTHandler.uploadData(
+      nftFormat,
+      storage,
+      image,
+      nftMetadata
+    )
+    return await nftManager.execute({
+      method: 'safeMint',
+      value,
+      params: [address, uri]
+    })
```

Comment:
You are checking that either of the two safeMint are being implemented but only safeMint(address,string) is used. Also tokenId is only used to decide which function to call but is not used in the transaction.

Reply:
updated